src/bio/bio_context.c

```diff
@@ -894,15 +973,17 @@ int bio_mc_close(struct bio_meta_context *bio_mc, enum bio_mc_flags flags)
 		if (rc1)
 			rc = rc1;
 	}
-	if (bio_mc->mc_meta) {
-		rc1 = bio_ioctxt_close(bio_mc->mc_meta, flags);
+	if (bio_mc->mc_wal) {
+		rc1 = bio_ioctxt_close(bio_mc->mc_wal, flags);
 		if (rc1 && !rc)
 			rc = rc1;
+		wal_close(bio_mc);
```

Comment:
wal_close() requires io context, it needs be called before closing wal io context. The closing order should be: wal_close(), meta_close(), wal & meta io context close.